### PR TITLE
fix: randomize all layer types

### PR DESCRIPTION
**Pull request by aski02**
References #18.

### CODE REVIEW

**Review by dilyabareeva — CHANGES_REQUESTED** [review]
Hi @aski02. Thanks for your work. I have two major comments:

1) The tests are only checking if parameters have been changed after randomization. Could we add tests that do inference on a batch of random data and see if the model output NaNs? That was the main reason we limited randomization to Linear layers previously.
2) As far as I remember, the current approach didn't work for [BatchNorm](https://pytorch.org/docs/main/generated/torch.nn.BatchNorm1d.html#torch.nn.BatchNorm1d) layers. Could we add tests for them as well?

File: tests/metrics/test_heuristics_metrics.py [hunk 1]
```diff
@@ -91,6 +101,43 @@ def test_randomization_metric(
...
+    with torch.no_grad():
+        original_out = model(random_tensor)
+        randomized_out = rand_model(random_tensor)
+
+    assert not torch.allclose(
+        original_out, randomized_out
+    ), "Outputs do not differ after randomization"
+    assert not torch.isnan(
+        randomized_out
+    ).any(), "Randomized model output contains NaNs."
```
Comment (dilyabareeva) [inline tests/metrics/test_heuristics_metrics.py:122]: Could we make 2) and 3) separate tests? In general, we aim to have only 1 assert per test. This makes debugging easier.

File: tests/metrics/test_heuristics_metrics.py [hunk 2]
```diff
@@ -67,17 +69,25 @@ def test_randomization_metric(
     tmp_path,
     request,
 ):
```
Comment (dilyabareeva) [inline tests/metrics/test_heuristics_metrics.py:71]: Can we add some additional (tiny) architectures to conftest.py, and add a couple of parameterizations (extra items in the parametrization list) with these architectures? Maybe the easiest would be to create those tiny architectures manually, and not train them.

**Review by dilyabareeva — CHANGES_REQUESTED** [review]
Great work @aski02, I have added some comments :)

**Review by dilyabareeva — APPROVED** [review]
Hi @aski02. Thanks a lot, great work. I've added a small commit and will merge now.

Please try to rely more on test parametrization instead of creating new tests wherever possible :)
